UI/src/app/components/panel/panel-base.component.ts

```diff
@@ -117,9 +112,48 @@ export class PanelComponent implements OnInit, AfterViewInit {
     }
   }
 
-  setExpandedFlagOnSourceAndTargetTables(table: ITable, expanded: boolean) {
-    this.stateService.state[table.area].tables
-      .filter(t => t.id === table.id)
-      .forEach(t => (t.expanded = expanded));
+  setExpandedFlagOnSourceAndTargetTables(expanded: boolean) {
+    this.storeService.state[this.area].filter(t => t.id === this.table.id).forEach(t => (t.expanded = expanded));
+  }
+
+  onCheckboxChange(event: MatCheckboxChange) {
+    for (const row of this.table.rows) {
+      const connections = this.bridgeService.findCorrespondingConnections(this.table, row);
+      for (const connection of connections) {
+        let action;
+        if (!event.checked) {
+          action = this.unLinkFields;
+        } else {
+          action = this.linkFields;
+        }
+
+        this.similarFieldsAction(connection, action.bind(this));
+      }
+    }
+  }
+
+  similarFieldsAction(connection, action) {
+    this.tables.forEach(table => {
+      if (table.name !== this.table.name) {
```

Comment:
As an option to decrease nesting levels
We can use early return there:
```ts
if (table.name === this.table.name) {
   return;
}
```

Reply:
done